lineman/app/components/dashboard_page/dashboard_page_controller.coffee

```diff
@@ -1,97 +1,60 @@
-angular.module('loomioApp').controller 'DashboardPageController', ($rootScope, Records, CurrentUser, LoadingService) ->
+angular.module('loomioApp').controller 'DashboardPageController', ($rootScope, Records, CurrentUser, LoadingService, ThreadQueryService) ->
   $rootScope.$broadcast('currentComponent', 'dashboardPage')
   $rootScope.$broadcast('setTitle', 'Dashboard')
 
-  @loaded =
-    sort_by_date:
-      show_all: 0
-      show_unread: 0
-      show_proposals: 0
-    sort_by_group:
-      show_all: 0
-      show_unread: 0
-      show_proposals: 0
-  @perPage =
-    sort_by_date: 25
-    sort_by_group: 10
-  @groupThreadCounts =
-    hidden:    0
-    collapsed: 5
-    expanded:  10
-
-  @sort   = -> CurrentUser.dashboardSort
-  @filter = -> CurrentUser.dashboardFilter
-
-  @loadParams = ->
-    filter: @filter()
-    per:    @perPage[@sort()]
-    from:   @loadedCount()
-
-  @loadedCount = (group) =>
-    if group
-      @groupThreadCounts[group.dashboardStatus or 'collapsed']
-    else
-      @loaded[@sort()][@filter()]
+  Records.votes.fetchMyRecentVotes()
 
-  @loadMore = (options = {}) =>
-    @loaded[@sort()][@filter()] = @loadedCount() + @perPage[@sort()]
-    switch @sort()
-      when 'sort_by_date'  then Records.discussions.fetchInboxByDate(@loadParams())
-      when 'sort_by_group' then Records.discussions.fetchInboxByGroup(@loadParams())
+  @perPage = 25
+  @loaded =
+    show_all:           0
+    show_muted:         0
+    show_proposals:     0
+    show_participating: 0
+  @filter    = -> CurrentUser.dashboardFilter
+  @nowLoaded = -> @loaded[@filter()]
+
+  @timeframes =
+    today:     { from: '1 second ago', to: '-10 year ago' }
+    yesterday: { from: '1 day ago',    to: '1 second ago' }
+    thisweek:  { from: '1 week ago',   to: '1 day ago' }
+    thismonth: { from: '1 month ago',  to: '1 week ago' }
+    older:     { from: '3 month ago',  to: '1 month ago' }
+  @timeframeNames = _.map @timeframes, (timeframe, name) -> name
+  @timeframeQueryFor = (name) -> @timeframes[name].view
+
+  @groupThreadLimit = 5
+  @groups = -> CurrentUser.parentGroups()
+  @groupName = (group) -> group.name
+  @groupQueryFor = (group) -> @["group#{group.id}"]
+  @moreForThisGroup = (group) -> @groupQueryFor(group, { filter: @filter() }).length() > @groupThreadLimit
+
+  @updateQueries = ->
```

Comment:
We now calculate these views on large page events (changing a filter, loading more records, navigating to the page, etc.), and they (the _views_, not the collections) are static until the next large page event. These queries are the objects we pass to the underlying thread collection 

Timeframes are stored in
```
this.timeframes.today.view
this.timeframes.yesterday.view
etc
```

Groups are stored in
```
this.group45
this.group600
```
...but that's not great, so I'm going to namespace em in my next pass through this.
(I'm probably going to change this to group.key too, instead of group.id)